Dropdown toolbar (#179)
pxblue/react-component-library: docs/DropdownToolbar.md

# Dropdown Toolbar

> The `<DropdownToolbar>` component is used to display a toolbar with a dropdown.

Reviewer: ```suggestion
The `<DropdownToolbar>` component is used to display a toolbar with a dropdown menu for a subtitle.
```

Reviewer: May also want to specify something about you can pass in a list of menuitems to render or your own custom menu content if you want more control.

<div style="width: 100%; text-align: center">
    <img width="100%" style="max-width: 600px" alt="Dropdown Toolbar" src="./images/dropdownToolbar.png">
</div>

## Usage

```tsx
import { DropdownToolbar } from '@pxblue/react-components';
...
<DropdownToolbar title={"Title"} subtitleLabel={"Subtitle"} menuItems={[
  {
    label: "Menu Item 1";
    onClick: () => {};
  },
  {
     label: "Menu Item 2";
    onClick: () => {};
  },
  {
     label: "Menu Item 3";

>     onClick: () => {};

Reviewer: indentation misaligned for onClick

  }
]}/>
```

> ### API

Reviewer: ```suggestion
## API
```

<div style="overflow: auto;">

| Prop Name      | Description                                    | Type                     | Required | Default   |
| -------------- | ---------------------------------------------- | ------------------------ | -------- | --------- |
| classes        | Style Overrides                                | `DropdownToolbarClasses` | no       |           |
| customMenu     | Custom content to be displayed in the menu     | `JSX.Element`            | no       |           |
| menuItems      | Items to be displayed in the menu              | `ToolbarMenuItem[]`      | no       |           |
| navigationIcon | Navigation Icon to be displayed                | `JSX.Element`            | no       |           |
| subtitleLabel  | Subtitle text to be displayed                  | `string` | `undefined`   | yes      |           |
| title          | Title text to be displayed                     | `string`                 | yes      |           |

</div>

> 

Reviewer: Missing the 'Any other props supplied will be provided to the root element...' blurb

> #### ToolbarMenuItem

Reviewer: ```suggestion
### ToolbarMenuItem
```

| Prop Name | Description                                    | Type                   | Required | Default   |
| --------- | ---------------------------------------------- | ---------------------- | -------- | --------- |
| label     | Label used for a menu item                     | `string`               | yes      |           |
| onClick   | Function to call when menu item is clicked     | `Function`             | yes      |           |

> #### Classes

Reviewer: ```suggestion
### Classes
```

You can override the classes used by PX Blue by passing a `classes` prop. It supports the following keys:

| Name                  | Description                          |
| --------------------- | ------------------------------------ |
| root                  | Styles applied to the root element   |
| menuItem              | Styles applied to the menu item      |

> | navigationIconWrapper | Styles applied to the navigationIcon |

Reviewer: Let's avoid naming things with 'wrapper' in our React components. In the DrawerHeader, we just call this `navigation`

</div>

Any other props supplied will be provided to the root element (`Toolbar`).

### ToolbarMenuItem

| Prop Name | Description                                    | Type                   | Required | Default   |
| --------- | ---------------------------------------------- | ---------------------- | -------- | --------- |
| label     | Label used for a menu item                     | `string`               | yes      |           |
| onClick   | Function to call when menu item is clicked     | `Function`             | yes      |           |

### Classes

You can override the classes used by PX Blue by passing a `classes` prop. It supports the following keys:

| Name                  | Description                                            |
| --------------------- | ------------------------------------------------------ |
| root                  | Styles applied to the root element                     |
| menuItem              | Styles applied to the menu item                        |
| navigation            | Styles applied to the navigationIcon                   |
| subtitle              | Styles applied to the subtitle text                    |

> | subtitleContent       | Styles applied to subtitle and arrowDropdown container |

Reviewer: ```suggestion
| subtitleContent       | Styles applied to subtitle and dropdown arrow container |
```
